storage_controller/src/service.rs

```diff
@@ -2744,6 +2745,115 @@ impl Service {
         Ok(timeline_info)
     }
 
+    pub(crate) async fn tenant_timeline_detach_ancestor(
+        &self,
+        tenant_id: TenantId,
+        timeline_id: TimelineId,
+    ) -> Result<models::detach_ancestor::AncestorDetached, ApiError> {
+        tracing::info!("Detaching timeline {tenant_id}/{timeline_id}",);
+
+        let _tenant_lock = trace_shared_lock(
+            &self.tenant_op_locks,
+            tenant_id,
+            TenantOperations::TimelineDetach,
+        )
+        .await;
```

Comment:
This can share the lock with timeline deletion (`Service::tenant_timeline_delete`). Is that ok?

Reply:
Yes, it should be, haven't yet written a test for it though.

Reply:
Aah you probably meant that there could also be a timeline delete for the detached timeline. That should be all-ok as well, cancel fast, and allow deletion to proceed. No test yet, but it's on the list.

Reply:
Oke sounds good. Feel free to resolve since you're planning on testing this. No need to have it in this pr.